x-pack/plugin/logsdb/qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/KeywordRollingUpgradeIT.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.upgrades;
+
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.test.rest.ObjectPath;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import static org.elasticsearch.upgrades.AbstractStringTypeRollingUpgradeIT.createTemplate;
+import static org.elasticsearch.upgrades.AbstractStringTypeRollingUpgradeIT.formatInstant;
+import static org.elasticsearch.upgrades.AbstractStringTypeRollingUpgradeIT.getIndexSettingsWithDefaults;
+import static org.elasticsearch.upgrades.AbstractStringTypeRollingUpgradeIT.startTrial;
+import static org.elasticsearch.upgrades.LogsdbIndexingRollingUpgradeIT.assertDataStream;
+import static org.elasticsearch.upgrades.StandardToLogsDbIndexModeRollingUpgradeIT.enableLogsdbByDefault;
+import static org.elasticsearch.upgrades.StandardToLogsDbIndexModeRollingUpgradeIT.getWriteBackingIndex;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.notNullValue;
+
+public class KeywordRollingUpgradeIT extends AbstractRollingUpgradeWithSecurityTestCase {
```

Comment:
After you merge in main, this test needs to be changed. The test needs to upgrade test cluster from the test method instead of relying on super class.

Reply:
ahhh, thanks Jordan! This makes sense now